experiments/GCM/heldsuarez.jl

```diff
@@ -0,0 +1,140 @@
+using LinearAlgebra
+using StaticArrays
+using Test
+using CLIMA
+using CLIMA.Atmos
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.Mesh.Filters
+using CLIMA.Mesh.Grids
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+using CLIMA.VariableTemplates
+
+
+const ArrayType = CLIMA.get_array_type()
+
+Base.@kwdef struct HeldSuarezSetup{FT}
+    p_ground::FT = MSLP
+    T_initial::FT = 255
+    domain_height::FT = 30e3
+end
+
+function (setup::HeldSuarezSetup)(state, aux, coords, t) 
+    # callable to set initial conditions
+    FT = eltype(state)
+
+    r = norm(coords, 2)
+    h = r - FT(planet_radius)
+
+    scale_height = R_d * setup.T_initial / grav
+    p            = setup.p_ground * exp(-h / scale_height)
+
+    state.ρ  = air_density(setup.T_initial, p)
+    state.ρu = SVector{3, FT}(0, 0, 0)
+    state.ρe = state.ρ * (internal_energy(setup.T_initial) + aux.orientation.Φ)
+
+    return nothing
+end
+
+function config_heldsuarez(FT, N, resolution, setup)
+    config = CLIMA.GCM_Configuration("HeldSuarez", FT, ArrayType, setup,
+                                     ref_state=HydrostaticState(IsothermalProfile(setup.T_initial), FT(0)),
+                                     turbulence=ConstantViscosityWithDivergence(FT(0)),
```

Comment:
For `turbulence`, we'd want to use an anisotropic Smagorinsky filter (until we have horizontal hyperdiffusion). Perhaps @akshaysridhar can weigh in?